#type_of_cam:
#0---stereo kitti
#1---stereo euroc
#2---others (not available yet)

type_of_cam: 0

image_width: 1226
image_height: 370

T_init:
[1, 0, 0, 0,
 0, 1, 0, 0,
 0, 0, 1, 0,
 0, 0, 0, 1]

#imu, not used
T_world_body:
[ 1.0,  0.0,  0.0,  0.0,
  0.0,  1.0,  0.0,  0.0,
  0.0,  0.0,  1.0,  0.0,
  0.0,  0.0,  0.0,  1.0]

#cam0
cam0_intrinsics: [707.0912, 707.0912, 601.8873, 183.1104] #fx fy cx cy
cam0_distortion_coeffs: [0, 0, 0, 0] #k1 k2 r1 r2
T_body_cam0:
[ 1.0,  0.0,  0.0,  0.0,
  0.0,  1.0,  0.0,  0.0,
  0.0,  0.0,  1.0,  0.0,
  0.0,  0.0,  0.0,  1.0]
#[-0.00185773938524100, -0.00648146582601100, 0.999977309828700, 0.333504767534810,
#-0.999965951351000, 0.00805186015113400, -0.00180552862766100, -0.00479610810716563,
#-0.00803997520451600, -0.999946608177400, -0.00649620353613900, -0.0755785517707151,
#0, 0, 0, 1]

T_cam0_cam1:
[1, 0, 0, 0.537150653267924,
0, 1, 0, 0,
0, 0, 1, 0,
0, 0, 0, 1]

#reserved para for future functions
para_1: 0
para_2: 0
para_3: 0
para_4: 0

# Original File of Seq 00 - 02:
# P0: 7.188560000000e+02 0.000000000000e+00 6.071928000000e+02 0.000000000000e+00 0.000000000000e+00 7.188560000000e+02 1.852157000000e+02 0.000000000000e+00 0.000000000000e+00 0.000000000000e+00 1.000000000000e+00 0.000000000000e+00
# P1: 7.188560000000e+02 0.000000000000e+00 6.071928000000e+02 -3.861448000000e+02 0.000000000000e+00 7.188560000000e+02 1.852157000000e+02 0.000000000000e+00 0.000000000000e+00 0.000000000000e+00 1.000000000000e+00 0.000000000000e+00
# P2: 7.188560000000e+02 0.000000000000e+00 6.071928000000e+02 4.538225000000e+01 0.000000000000e+00 7.188560000000e+02 1.852157000000e+02 -1.130887000000e-01 0.000000000000e+00 0.000000000000e+00 1.000000000000e+00 3.779761000000e-03
# P3: 7.188560000000e+02 0.000000000000e+00 6.071928000000e+02 -3.372877000000e+02 0.000000000000e+00 7.188560000000e+02 1.852157000000e+02 2.369057000000e+00 0.000000000000e+00 0.000000000000e+00 1.000000000000e+00 4.915215000000e-03
# Tr: 4.276802385584e-04 -9.999672484946e-01 -8.084491683471e-03 -1.198459927713e-02 -7.210626507497e-03 8.081198471645e-03 -9.999413164504e-01 -5.403984729748e-02 9.999738645903e-01 4.859485810390e-04 -7.206933692422e-03 -2.921968648686e-01

# http://www.cvlibs.net/datasets/kitti/setup.php
# use matlab to extract the info for yaml
# Sequence_id = '00';
# calib_path = fullfile('.\kitti\dataset\sequences',sequence_id,'calib.txt');
## Camera Info
# calib_data = dlmread(calib_path,' ',0,1);
# P0 = reshape(calib_data(1,:),[4,3])';
# K = P0(:,1:3);
# K_cam0 = K_cam1 = K
# cam to body
# T_body_cam0 = TransformInverse([Tr_velo_to_cam; 0 0 0 1]);
# P1 = reshape(calib_data(2,:),[4,3])';
# Tr_velo_to_cam = reshape(calib_data(5,:),[4,3])'; % Tr data in calib files
# R_cam1 = eye(3);
# t_cam1 = pinv(K) * (P0(:,4) - P1(:,4));
# T_c0_c1 = [R_cam1,t_cam1; 0 0 0 1];

# convert images to rosbag:
# https://github.com/JotellyBarros/image2rosbag_KITTIodometry
